src/Enterspeed.Source.UmbracoCms.V9/Data/Migration/AddContentStateToJobsTable.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Enterspeed.Source.UmbracoCms.V9.Data.Models;
+using Enterspeed.Source.UmbracoCms.V9.Data.Schemas;
+using Umbraco.Cms.Infrastructure.Migrations;
+
+namespace Enterspeed.Source.UmbracoCms.V9.Data.Migration
+{
+    public class AddContentStateToJobsTable : MigrationBase
+    {
+        public AddContentStateToJobsTable(IMigrationContext context)
+            : base(context)
+        {
+        }
+
+        protected override void Migrate()
+        {
+            var jobsTable = Database.PocoDataFactory.ForType(typeof(EnterspeedJobSchema));
+            var jobsTableName = jobsTable.TableInfo.TableName;
+
+            var entityTypeColumnName = "ContentState";
+
+            // If column doesnt exist, drop table and recreate it
+            if (!ColumnExists(jobsTableName, entityTypeColumnName))
+            {
+                Create
+                    .Column("ContentState")
+                    .OnTable(jobsTableName)
+                    .AsInt32().NotNullable().WithDefaultValue(EnterspeedContentState.Publish.GetHashCode())
```

Comment:
How this will work in scenario where user before upgrading to newest connector, has failed job in the table, which is not cleaned up yet. During migration would it fail, since new column is not nullable?

Reply:
@msoelvsteen any inputs on this one?